File: C++/binarySearch.cpp
#include<iostream>
#include<bits/stdc++.h>
using namespace std;
int s;
int binarySearch(int arr[],int i, int j)
{
    int a=1,b=0;
    int mid=(i+j)/2;
    while(i<j)
    {
        if(arr[mid]==s)
        {
            return a;
        }
        if(s>arr[mid])
        {
            return binarySearch(arr,mid+1,j);
        }
        if(s<arr[mid])
        {
            return binarySearch(arr,i,mid);
        }
    }
    return b;
}
int main()
{
    int n,result;
    cout<<"enter size of array: ";
    cin>>n;
    int arr[n];
    cout<<"enter array elements: "<<endl;
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    sort(arr,arr+n);
    cout<<"enter element to search: ";
    cin>>s;
    result=binarySearch(arr,0,n-1);
    if(result==1)
        cout<<"element is present";
    else
        cout<<"element is not present";
}

File: Java/BinarySearch.java
import java.util.Scanner;
public class BinarySearch {
    public static void main(String[] args) {
        int arr [] = {-3,-2,0,1,3,6,16,18,45,89,101,108};
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter an integer to search it in the array :");
        int target = sc.nextInt();
        int indexNumber = binarySearch(arr,target);
        printMessage(target, indexNumber);
    }

    static int binarySearch(int [] arr , int key){
        int start = 0;
        int end = arr.length-1;
        while(start <=end){

            int mid = start + (end - start)/2;

            if(key>arr[mid]){
                start = mid+1;
            }
            else if(key<arr[mid]){
                end = mid-1;
            }
            else{
                return mid;
            }
        }
        return -1;
    }
    static void printMessage(int target, int indexnumber){ // This method prints the index of target when called.
        if(indexnumber == -1){
            System.out.println(target + " is not found in the array.");
        }
        else{
            System.out.println(target + " is found in the array at index " + indexnumber);
        }
    }
}
